### PR TITLE
feat: generated npm_link_targets() macro that lists links generated by npm_link_all_packages()

### PR DESCRIPTION
**Pull request by gregmagolan**
Resolves https://github.com/aspect-build/rules_js/issues/933

### CODE REVIEW

**Comment by gregmagolan** [comment]
cc @Aghassi

**Comment by jbedard** [comment]
Is it possible to have a test with one of these files checked in and asserting it doesn't change? Both for the test but mainly just as an example of what is being generated, seeing the diff when it changes etc.

**Comment by jbedard** [comment]
Or maybe just add a few `npm_link_targets()` calls to 1+ of the e2e tests that contain workspaces? Can we do some assertions on those within BUILDs?

File: npm/private/npm_translate_lock_generate.bzl [hunk 1]
```diff
@@ -752,7 +758,26 @@ sh_binary(
...
 load("@aspect_rules_js//npm/private:npm_package_store.bzl", _npm_package_store = "npm_package_store")""")
 
-    defs_bzl_body = [
+    npm_link_targets_bzl = [
+        """def npm_link_targets(name = "node_modules", package = None):
+    \"\"\"Generated list of target names that are linked by npm_link_all_packages()
+
+    Args:
+        name: name of catch all target to generate for all packages linked
+        package: Bazel package to generate targets names for. If unspecified, the current package is used.
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply npm/private/npm_translate_lock_generate.bzl]: 👍

**Review by kormide — APPROVED** [review]

**Review by Aghassi — APPROVED** [review]
:shipit:

**Pull request merged**